File: keeper/src/main/java/com/github/abhijitpparate/keeper/screen/login/presenter/registration/RegistrationContract.java
package com.github.abhijitpparate.keeper.screen.login.presenter.registration;

import android.support.annotation.StringRes;

public interface RegistrationContract {

    interface View {
        String getName();
        String getEmail();
        String getPassword();
        String getPasswordConfirmation();

        void showLoginScreen();
        void showHomeScreen();
        void showProgressIndicator(boolean show);

        void setPresenter(RegistrationContract.Presenter presenter);
        void makeToast(@StringRes int strId);
        void makeToast(String message);
    }

    interface Presenter {

        void onLoginClick();
        void onRegisterClick();

        void subscribe();
        void unsubscribe();
    }

}

File: keeper/src/main/java/com/github/abhijitpparate/keeper/screen/login/presenter/registration/RegistrationContract.kt
package com.github.abhijitpparate.keeper.screen.login.presenter.registration

import android.support.annotation.StringRes

interface RegistrationContract {

    interface View {
        val name: String
        val email: String
        val password: String
        val passwordConfirmation: String

        fun showLoginScreen()
        fun showHomeScreen()
        fun showProgressIndicator(show: Boolean)

        fun setPresenter(presenter: RegistrationContract.Presenter)
        fun makeToast(@StringRes strId: Int)
        fun makeToast(message: String)
    }

    interface Presenter {

        fun onLoginClick()
        fun onRegisterClick()

        fun subscribe()
        fun unsubscribe()
    }

}
